osu.Game/Screens/Select/Leaderboards/Leaderboard.cs

```diff
@@ -38,15 +50,38 @@ public IEnumerable<Score> Scores
             set
             {
                 scores = value;
-                getScoresRequest?.Cancel();
 
-                scrollFlow?.FadeOut(200);
-                scrollFlow?.Expire();
+                placeholderContainer.FadeOut(fade_duration);
+
+                scrollFlow?.FadeOut(fade_duration).Expire();
                 scrollFlow = null;
 
+                loading.Hide();
+
                 if (scores == null)
                     return;
 
+                if (!scores.Any())
+                {
+                    placeholderFlow.Children = new Drawable[]

```

Comment:
instead of setting children here, just make:

```
private class NoScoresPlaceholder : Container (or FlowContainer or whatever)
{
    ctor { Children = { SpriteIcon, OsuSpriteText } }
}

private class RetrievalFailPlaceholder : Container (or FlowContainer or whatever)
{
    ctor { Children = { RetryIcon, OsuSpriteText } }
}
```

and change this code to

```
placeholderContainer.Clear(true);
placeholderContainer.Child = new NoScoresPlaceholder();
placeholderContainer.FadeInFromZero(500, EasingTypes.OutQuint);
```
or so

to tidy things up